src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs

```diff
@@ -701,6 +743,48 @@ public virtual int ReadByte()
             return r == 0 ? -1 : oneByteArray[0];
         }
 
+        public void ReadExactly(Span<byte> buffer) =>
+            _ = ReadAtLeastCore(buffer, buffer.Length, throwOnEndOfStream: true);
```

Comment:
Both `ReadAtLeast` and `ReadExactly` can never read more bytes than the buffer that is passed into it.

So if you have a `buffer = new byte[10]`, and call `ReadAtLeast(buffer, 5)`, the Read call will read between `5-10` bytes of data. It is guaranteed to be `minimumBytes <= numRead` and `numRead <= buffer.Length`.

So yes, if you had "real data" in the buffer beyond the `minimumBytes` index, it is possible for that data to be overwritten by the Read call. You would need to slice the buffer down in this situation.  The same goes for `Read(Span<byte>)`, it can read to fill up the entire buffer.